### PR TITLE
Add Scilab 6.0.2 and 6.1.0 from Ubuntu repos

### PR DESCRIPTION
**Pull request by davidcl**
1. setup travis using the SCI env. variable
  2. detect includes using the SCI env. variable
  3. log Scilab version at ./configure

Note: the Ubuntu Scilab version is preserved as a reference package

### CODE REVIEW

**Comment by wsfulton** [comment]
Thanks, nearly okay. Some points to address:

1. The two versions which you put under `allow_failures` must also be under the normal section, that's why they weren't run.
2. Version 5.5.2 is run twice. Look for "showing scilab version" in the output logs.
3. In configure.ac you are looking for an env variable called `SCI`. I suspect this is because you explicitly exported in the travis-linux-install.sh scripts. These feels rather odd unless setting SCI is the normal way to install Scilab. Can you clarify and comment the install if it is. How it should work is configure.ac should search for standard installs of Scilab. travis-linux-install.sh should install Scilab in the standard way (following some instructions from Scilab). We try and install pre-built installations normally, hence we normally install from an apt ppa repository for versions that don't come with Ubuntu 16.04.

**Comment by davidcl** [comment]
>     1. The two versions which you put under `allow_failures` must also be under the normal section, that's why they weren't run.
Thanks for the information, I will add them.
 
>     2. Version 5.5.2 is run twice. Look for "showing scilab version" in the output logs.
Yep, as `scilab` is provided through apt I preserved it for reference. What do you prefer, only use binaries from www.scilab.org ? or use the apt shipped version and the 2 other binaries from www.scilab.org ?

>     3. In configure.ac you are looking for an env variable called `SCI`. I suspect this is because you explicitly exported in the travis-linux-install.sh scripts. These feels rather odd unless setting SCI is the normal way to install Scilab. Can you clarify and comment the install if it is.
SCI is a commonly known Scilab variable that could also be used as an env. variable to select a specific directory to start from (this is detected by the launcher script). I could also detect it from the binary fullpath but it might requires more code on the ./configure

> How it should work is configure.ac should search for standard installs of Scilab. travis-linux-install.sh should install Scilab in the standard way (following some instructions from Scilab). We try and install pre-built installations normally, hence we normally install from an apt ppa repository for versions that don't come with Ubuntu 16.04.
I understood that however this pull-request is to use the stock binary from upstream that might ease cross-version checking. If you think that having the apt source is good enough for the CI, no problem I will keep that patchset in the [Scilab fork](https://github.com/scilab/swig) to work on multiple version support. 

Thanks for your answer

**Comment by wsfulton** [comment]
> What do you prefer, only use binaries from www.scilab.org ? or use the apt shipped version and the 2 other binaries from www.scilab.org ?
The latter - distro scilab for Ubuntu Xenial (installed using apt) plus whatever other newer versions you need to test. Looks like you are still testing 5.5.2 twice, so please remove the `VER=5.5.2` build.

File: Examples/Makefile.in [hunk 1]
```diff
@@ -1381,14 +1381,15 @@ r_clean:
 SCILAB = @SCILAB@
 SCILAB_INC= @SCILABINCLUDE@
 SCILAB_OPT = @SCILABOPT@
+SCILAB_VERSION = @SCILAB_FULL_VERSION@
```
Comment (wsfulton) [inline Examples/Makefile.in]: `-targetversion` should take the 'major version', so suggest:
```suggestion
SCILAB_MAJOR_VERSION = @SCILAB_MAJOR_VERSION@
```
plus other necessary changes.

File: configure.ac [hunk 2]
```diff
@@ -1238,6 +1240,7 @@ fi
 AC_SUBST(SCILAB)
 AC_SUBST(SCILABINCLUDE)
 AC_SUBST(SCILABOPT)
+AC_SUBST(SCILAB_FULL_VERSION)
```
Comment (wsfulton) [inline configure.ac]:
```suggestion
AC_SUBST(SCILAB_MAJOR_VERSION)
```

**Comment by wsfulton** [comment]
> SCI is a commonly known Scilab variable that could also be used as an env. variable to select a specific directory to start from (this is detected by the launcher script)
Can you add a reference to Scilab docs for this please as I don't see this as being particularly commonly known amongst SWIG developers.

**Comment by ojwb** [comment]
Thanks to the travispocalypse, we've had to switch to using github actions so unfortunately the CI part of this is going to need redoing.

Looks like we currently only test one scilab version with github actions - seems to be the default is on ubuntu 18.04 which is 6.0.1.

**Comment by ojwb** [comment]
```
$ rmadison -uubuntu scilab
 scilab | 5.5.0-2                | trusty/universe          | source, all
 scilab | 5.5.2-2ubuntu3         | xenial/universe          | source, all
 scilab | 6.0.1-1ubuntu1         | bionic/universe          | source, all
 scilab | 6.0.1-7ubuntu1~18.04   | bionic-security/universe | source, all
 scilab | 6.0.1-7ubuntu1~18.04.2 | bionic-updates/universe  | source, all
 scilab | 6.1.0+dfsg1-1ubuntu3   | focal/universe           | source, all
 scilab | 6.1.0+dfsg1-1ubuntu3.1 | focal-updates/universe   | source, all
 scilab | 6.1.0+dfsg1-7          | impish/universe          | source, all
 scilab | 6.1.0+dfsg1-7          | jammy/universe           | source, all
 scilab | 6.1.1+dfsg2-3          | jammy-proposed/universe  | source
```

Gitlhub actions currently supports bionic (`os: ubuntu-18.04`) and focal (`os: ubuntu-20.04`) so we could add a CI build with 6.1.0 easily.

This PR is approaching 2 years old - Is scilab 5.5 still relevant?

If so we either need to see if you can specify a custom container image for xenial in github actions, or rework this PR for github actions.

**Comment by ojwb** [comment]
There's still a download link for 5.5.x on the scilab download page so it seems scilab upstream thinks it's still relevant (or are slack at updating their website).

Enabling 6.1.0 on a branch shows it doesn't currently work: https://github.com/swig/swig/actions/runs/1780541161

So to summarise:

* This patch needs reworking for github actions
* Scilab 5.5.x would need installing from upstream or something
* Scilab 6.0.1 is already covered by github actions
* SWIG needs fixing to support Scilab 6.1.0

**Comment by davidcl** [comment]
@ojwb the Scilab 5.5 is legacy but is still use for some toolboxes/historical codes. Most of the users switched to Scilab 6.0 and to Scilab 6.1 now.

Thanks to your #scilab-debug-ci branch, I move the logic to #1bb7f00d9a3fc33bce68dc9c3f7d0e9031ac1a32 ease the merge of both SWIG logic and github actions for Scilab 6.1.

**Comment by ojwb** [comment]
This PR reintroduces the deleted files `.travis.yml` and `Tools/travis-linux-install.sh` and adds use of the `-targetversion` option which #1751 removes.

There also seems a significant overlap in purpose and files modified with #1751.  Should we really have both?  If so, which order should they be reviewed and merged in?

**Comment by wsfulton** [comment]
@davidcl feel free to rebase this branch to address @ojwb points. Please include the absolute minimum for adding in the additional Github Actions testing.

**Comment by davidcl** [comment]
> There also seems a significant overlap in purpose and files modified with #1751. Should we really have both? If so, which order should they be reviewed and merged in?

You are correct, there is a significant overlap and IMHO this pull request can be close in favor of #1751 that pass the CI.

**Pull request closed without merging**